node/src/types/block.rs

```diff
@@ -726,6 +726,11 @@ impl Block {
         self.proofs.push(proof)
     }
 
+    /// Returns true if block already contains the proof.
+    pub(crate) fn contains_proof(&self, proof: &Signature) -> bool {
```

Comment:
Isn't it more natural to look that up by public key, rather than by signature?
Especially with non-unique signature schemes: It probably doesn't make sense to add two different signatures by the same validator.

Reply:
Hmm, but a block contains only a `Signature` … I guess we should add public key to that as well. Otherwise, there's no way of verifying whether proofs even increase security at all (whether block was signed by at-that-time bonded validator).

Reply:
OK… since the block structure is temporary anyway, and we'll remove the signatures from the block itself, I'm totally fine with leaving it as it is for now!